multiapps-controller-core/src/main/java/org/cloudfoundry/multiapps/controller/core/auditlogging/ConfigurationEntryServiceAuditLog.java

```diff
@@ -0,0 +1,72 @@
+import org.cloudfoundry.multiapps.controller.persistence.model.ConfigurationEntry;
+
+import java.text.MessageFormat;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+public class ConfigurationEntryServiceAuditLog {
+
+    private static final String PROVIDER_ID_PROPERTY_NAME = "providerId";
+    private static final String PROVIDER_NID_PROPERTY_NAME = "providerNid";
+    private static final String PROVIDER_VERSION_PROPERTY_NAME = "providerVersion";
+    private static final String PROVIDER_NAMESPACE_PROPERTY_NAME = "providerNamespace";
+    private static final String PROVIDER_TARGET_PROPERTY_NAME = "providerTarget";
+    private static final String PROVIDER_CONTENT_PROPERTY_NAME = "providerContent";
+    private static final String PROVIDER_CONTENT_ID_PROPERTY_NAME = "providerContentId";
+
+    private final AuditLoggingFacade auditLoggingFacade;
+
+    public ConfigurationEntryServiceAuditLog(AuditLoggingFacade auditLoggingFacade) {
+        this.auditLoggingFacade = auditLoggingFacade;
+    }
+
+    public void logAddConfigurationEntry(String username, String spaceGuid, ConfigurationEntry entry) {
+        String performedAction = MessageFormat.format(Messages.ENTRY_CREATE, spaceGuid);
+        auditLoggingFacade.logConfigurationChangeAuditLog(new AuditLogConfiguration(username,
+                                                                                    spaceGuid,
+                                                                                    performedAction,
+                                                                                    Messages.ENTRY_CONFIG,
+                                                                                    createAddEntryConfigurationIdentifier(entry)),
+                                                          ConfigurationChangeActions.CONFIGURATION_CREATE);
+    }
+
+    public void logUpdateConfigurationEntry(String username, String spaceGuid, ConfigurationEntry oldEntry, ConfigurationEntry newEntry) {
+        String performedAction = MessageFormat.format(Messages.ENTRY_UPDATE, spaceGuid);
+
+        List<String> attributes = List.of(oldEntry.getConfigurationType(), JsonUtil.toJson(oldEntry),
+                                          JsonUtil.toJson(newEntry));
+
+        auditLoggingFacade.logConfigurationChangeAuditLog(new AuditLogConfiguration(username,
+                                                                                    spaceGuid,
+                                                                                    performedAction,
+                                                                                    Messages.ENTRY_CONFIG),
+                                                          ConfigurationChangeActions.CONFIGURATION_UPDATE, attributes);
+    }
+
+    private Map<String, String> createAddEntryConfigurationIdentifier(ConfigurationEntry entry) {
```

Comment:
Identifier is the wrong word for it - maybe rename to 'buildAddConfigEntryParameters' - parameters is consistent with the field in AuditLogConfiguration class; addConfigurationEntry is more correct than addEntryConfiguration and also shortening of configuration to config might be better for readibility